tests-e2e/argocd/argocd_appproject_test.go

```diff
@@ -0,0 +1,195 @@
+			}
+		}
+
+		BeforeEach(func() {
+			By("Delete old namespaces and kube-system resources")
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			ctx = context.Background()
+
+			config, err = fixture.GetSystemKubeConfig()
+			Expect(err).To(BeNil())
+
+			k8sClient, err = fixture.GetKubeClient(config)
+			Expect(err).To(BeNil())
+
+			appProject = appv1.AppProject{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-app-project",
+					Namespace: dbutil.GetGitOpsEngineSingleInstanceNamespace(),
+				},
+				Spec: appv1.AppProjectSpec{
+					Destinations: []appv1.ApplicationDestination{
+						{
+							Server:    "https://kubernetes.default.svc",
+							Namespace: fixture.GitOpsServiceE2ENamespace,
+						},
+					},
+				},
+			}
+
+			app = appv1.Application{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "argo-app-01",
+					Namespace: dbutil.GetGitOpsEngineSingleInstanceNamespace(),
+				},
+				Spec: appv1.ApplicationSpec{
+					Project: appProject.Name,
+					Source: appv1.ApplicationSource{
+						Path:           "resources/test-data/sample-gitops-repository/environments/overlays/dev",
+						TargetRevision: "HEAD",
+					},
+					Destination: appv1.ApplicationDestination{
+						Server:    "https://kubernetes.default.svc",
+						Namespace: fixture.GitOpsServiceE2ENamespace,
+					},
+					SyncPolicy: &appv1.SyncPolicy{
+						Automated: &appv1.SyncPolicyAutomated{},
+					},
+				},
+			}
+
+			secret = corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-app-project-secret",
+					Namespace: app.Namespace,
+					Labels: map[string]string{
+						"argocd.argoproj.io/secret-type": "repository",
+					},
+				},
+				Data: nil,
+				Type: "",
+			}
+
+			// Ensure there's no lingering app project or secret
+			cleanup()
+		})
+
+		AfterEach(func() {
+			// Ensure there's no lingering app project or secret
+			cleanup()
+		})
+
+		It("Should succeed if the app project and the application both use an https url", func() {
+			By("Creating an app project with an https url")
+			appProject.Spec.SourceRepos = []string{"https://github.com/redhat-appstudio/managed-gitops"}
+			Expect(k8sClient.Create(ctx, &appProject)).To(Succeed())
+
+			By("Creating an application with an https url")
+			app.Spec.Source.RepoURL = "https://github.com/redhat-appstudio/managed-gitops"
+			Expect(k8sClient.Create(ctx, &app)).To(Succeed())
+
+			Eventually(app, "2m", "1s").Should(
+				SatisfyAll(
+					appFixture.HaveSyncStatusCode(appv1.SyncStatusCodeSynced),
+					appFixture.HaveHealthStatusCode(health.HealthStatusHealthy),
+				))
+		})
+
+		// This test requires the following environment variable to be set.
+		//		GITHUB_SSH_KEY: The SSH key of the private GitHub repository
+		// If the variable is not set, the test will be skipped.
+		It("Should succeed if the app project and the application both use a git url", func() {
+			const giturl = "git@github.com:managed-gitops-test-data/private-repo-test.git"
```

Comment:
I'm 90% sure you already answered this in standup a little while ago, but it's always good to double check:
- I presume it's not possible to use 'git@github.com:redhat-appstudio/managed-gitops.git' here? (IIRC you said it's not)
- If my presumption is correct, could you add a quick little comment to the test that mentions that (that way the next person that comes to this test will know why we're doing it this way).

Reply:
I've added a comment to explain.